addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -286,58 +302,185 @@ else if ( topPath.endsWith( "/" ) )
     @Override
     public Transfer generateGroupFileContent( final Group group, final List<ArtifactStore> members, final String path,
                                               final EventMetadata eventMetadata )
-        throws IndyWorkflowException
+            throws IndyWorkflowException
+    {
+        final Metadata md = generateGroupMetadata( group, members, path );
+        if ( md != null )
+        {
+            String toMergePath = path;
+            if ( !path.endsWith( MavenMetadataMerger.METADATA_NAME ) )
+            {
+                toMergePath = normalize( normalize( parentPath( toMergePath ) ), MavenMetadataMerger.METADATA_NAME );
+            }
+
+            final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            Transfer target = null;
+            try
+            {
+                target = fileManager.getTransfer( group, toMergePath );
+                if ( target.exists() )
```

Comment:
If it's already on the filesystem, can we leave it there and just serve the Transfer? Then, we can delete the file when the group membership / member content changes and the file / metadata becomes obsolete?

Here, it looks like the code rewrites this file every time regardless of whether the metadata in the cache has changed since the last time it was written.

Reply:
What I consider it here for first idea is that we don't care about the old content of the temp file for the regeneration, so just delete it and create it again to make sure the content is always same to the Metadata content for insurance. But seems you are right because each time for a case which triggers a file deletion, it means the real metadata refresh case. If the temp file is till there, that means no metadata refresh cases happens and we can use the temp file directly. 

I'll change the logic and have a test later.